server/src/main/java/io/crate/metadata/functions/BoundSignature.java

```diff
@@ -25,10 +25,7 @@
 
 import io.crate.types.DataType;
 
-public final class BoundSignature {
-
-    private final List<DataType<?>> argTypes;
-    private final DataType<?> returnType;
+public record BoundSignature(List<DataType<?>> argTypes, DataType<?> returnType) {
 
```

Comment:
Note for reviewers: I needed an equals and a toString to make testing easier, so converting this to a record seemed easiest